storage-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java

```diff
@@ -50,97 +53,125 @@
+                final String prevFileName = prevPath.getName();
+                final Optional<ExternalCommitEntry> prevEntry = getExternalEntry(
+                    tablePath.toString(),
+                    prevFileName
+                );
+                if (prevEntry.isPresent() && !prevEntry.get().complete) {
+                    fixDeltaLog(fs, prevEntry.get());
                 } else {
-                    if(!fs.exists(prevPath)) {
+                    if (!fs.exists(prevPath)) {
                         throw new java.nio.file.FileSystemException(
                             String.format("previous commit %s doesn't exist", prevPath)
                         );
                     }
                 }
             } else {
-                ExternalCommitEntry entry = getExternalEntry(tablePath, path);
-                if(entry != null) {
-                    if(entry.complete && !fs.exists(path)) {
+                final String fileName = path.getName();
+                final Optional<ExternalCommitEntry> entry = getExternalEntry(
+                    tablePath.toString(),
+                    fileName
+                );
+                if (entry.isPresent()) {
+                    if (entry.get().complete && !fs.exists(path)) {
                         throw new java.nio.file.FileSystemException(
                             String.format(
-                                "Old entries for %s still exist in the database", tablePath
+                                "Old entries for table %s still exist in the external store",
+                                tablePath
                             )
                         );
                     }
                 }
             }
         }
-        String tempPath = createTemporaryPath(resolvedPath);
-        ExternalCommitEntry entry = new ExternalCommitEntry(
+
+        // Step 2: PREPARE the commit
+        final String tempPath = createTemporaryPath(resolvedPath);
+        final ExternalCommitEntry entry = new ExternalCommitEntry(
             tablePath,
             resolvedPath.getName(),
             tempPath,
-            false,  // not complete
-            null    // commitTime
+            false, // not complete
+            null // commitTime
         );
         writeActions(fs, entry.absoluteTempPath(), actions);
-        putExternalEntry(entry, false);
+        putExternalEntry(entry, false); // overwrite=false
 
         try {
+            // Step 3: COMMIT the commit to the delta log
             writeCopyTempFile(fs, entry.absoluteTempPath(), resolvedPath);
+
+            // Step 4: ACKNOWLEDGE the commit
             writePutCompleteDbEntry(entry);
-        } catch(Throwable e) {
-            LOG.info(String.format(
-                "%s: ignoring recoverable error: %s", e.getClass().getSimpleName(), e));
+        } catch (Throwable e) {
+            LOG.info(
+                "{}: ignoring recoverable error: {}", e.getClass().getSimpleName(), e
```

Comment:
for some reason in logs I see:
```
22/03/30 18:52:10 INFO BaseExternalLogStore: IOException: ignoring recoverable error: {}
java.io.IOException: injected failure: write_copy_temp_file
	at io.delta.storage.FailingDynamoDBLogStore.injectError(FailingDynamoDBLogStore.java:77)
	at io.delta.storage.FailingDynamoDBLogStore.writeCopyTempFile(FailingDynamoDBLogStore.java:52)

```

Reply:
notice `{}` placeholder in above message

Reply:
> Also whole stacktrace is present

Is that bad?

Reply:
Placeholder `{}` in logs is probably unexpected. Stacktrace is probably ok, I'm only asking if it is intentional

Reply:
good catch! thanks